scripts/with-docker-env

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# Make sure we always run from the root
+SCRIPTS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+cd "$SCRIPTS_DIR/.."
+
+#########
+# USAGE #
+#########
+
+function title() {
+    echo "Execute a command between starting and stopping the docker compose environment"
+}
+
+function usage() {
+    cat << EOF
+
+Usage:
+  $0 <subcommand>
+
+EOF
+}
+
+function help() {
+    cat << EOF
+
+Starts the docker compose environment, executes the subcommand and stops docker compose again.
+
+NOTE: This requires docker, docker-compose, a running dfx replica with II and the test app deployed and running II dev server (npm run start).
+EOF
+}
+
+scripts/start-selenium-env
+
+ret=0
+"$@" || ret="$?"
+
+echo "'$*' returned with $ret"
+
+cd docker-test-env
+docker compose down
```

Comment:
what happens if docker compose is not up? Could this already return with non-zero?

Reply:
If `docker compose` fails, you'll get the error output from docker and a non-zero exit code. E.g.:
```
$ scripts/with-docker-env npm run test:e2e-desktop
[+] Building 0.0s (0/0)
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?: driver not connecting
```

Reply:
Ah, and yes, failure of `docker compose down` will cause non-zero exit code.

Reply:
meaning the script will exit and you won't know what `$ret` was, correct? is it worth also "catching" the return code of `docker compose down` and then show both return codes before exiting?

Reply:
`echo "'$*' returned with $ret"` shows the return code?

Reply:
Added an explicit warning when `docker compose` fails to shut down properly.